### PR TITLE
fix: DPL MQTT topics not carrying the hostname, improve: DPL MQTT threshold settings

### PR DESCRIPTION
**Pull request by AndreasBoehm**
This fixes https://github.com/helgeerbe/OpenDTU-OnBattery/issues/963 by including the unique id of a DTU in the topics.
I also passed 'step' to home assistant for the voltages, and shortened the topic titles as requested here https://github.com/helgeerbe/OpenDTU-OnBattery/issues/737

**NOTE: This is a breaking change for existing users, as the id of the DPL in HA will change.**

The changes are based on https://github.com/helgeerbe/OpenDTU-OnBattery/blob/f0df583c13394b0f5e59e363e5ac77788a2663eb/src/MqttHandleHass.cpp#L331

### CODE REVIEW

**Comment by schlimmchen** [comment]
As far as I can tell, the "breaking" part of this PR is not too big of a deal. The respective topics are the same, only the entities for HomeAssistant auto discovery change. Does this mean workflows or automation for DPL values break?

**Comment by AndreasBoehm** [comment]
I need to re-test this as my home assistant installation is not working properly right now. I will let you know.

**Comment by AndreasBoehm** [comment]
This is how it looks in mqtt explorer now.
Screenshot 2024-06-03 at 09 07 00

**Comment by AndreasBoehm** [comment]
Its working now
Screenshot 2024-07-16 at 09 40 20

**Comment by AndreasBoehm** [comment]
Setting the thresholds works now without issues
Screenshot 2024-07-16 at 10 13 22

**Comment by AndreasBoehm** [comment]
@hawkeye80 you can install a build from [here](https://github.com/helgeerbe/OpenDTU-OnBattery/actions/runs/9953117258?pr=973) and check if two DPLs are now appearing in home assitant

**Comment by hawkeye80** [comment]
Flashed both DTUs and is looks fine.
image

**Comment by AndreasBoehm** [comment]
> Does this mean workflows or automation for DPL values break?
I am not 100% sure to be honest as i am not using any workflows or automations, will try to check what happens if i would.

**Comment by hawkeye80** [comment]
Yes, they are broken as new entities got created. Either automations and so on need to be adjusted or the prior used entity need to be renamed back.

**Comment by schlimmchen** [comment]
> Setting the thresholds works now without issues

Looking forward to this, nice!

> Yes, they are broken as new entities got created.

I say we label this as a breaking change, so it appears as such in the changelogs, and be done with it. Since this is about distinguishing different OpenDTUs, we simply cannot make one use the old scheme to have backwards-compatibility, while others use the new scheme. Even if multiple DTUs in the same network could find and talk to each other, who would use the old scheme.... That's nonsense. This PR is the way forward.

File: src/MqttHandlePowerLimiterHass.cpp [hunk 1]
```diff
@@ -143,22 +143,22 @@ void MqttHandlePowerLimiterHassClass::publishSelect(
 void MqttHandlePowerLimiterHassClass::publishNumber(
     const char* caption, const char* icon, const char* category,
     const char* commandTopic, const char* stateTopic, const char* unitOfMeasure,
-    const int16_t min, const int16_t max)
+    const int16_t min, const int16_t max, const double_t step)
```
Comment (schlimmchen) [inline src/MqttHandlePowerLimiterHass.cpp:146]: Please, don't use double_t nor double. Only float. Reason: Handling double precision floats is implemented in software, while floats can be handled in hardware.

Comment (AndreasBoehm) [reply src/MqttHandlePowerLimiterHass.cpp]: Thanks for teaching me more about the platform.
Changes pushed

**Comment by schlimmchen** [comment]
@AndreasBoehm Is this ready to be merged? I would like to squash the commits and use "BREAKING CHANGE: ..." as the commit message header, so it appears as such in the changelogs.

**Comment by AndreasBoehm** [comment]
From my point of view its ready to be merged, but please be aware that i did not fix all concerns that you mentioned in https://github.com/helgeerbe/OpenDTU-OnBattery/issues/737

Missing:
- really short names for the topic titles
- text label to set the mode

But right now i don't see that i will be working on this soon because i could not find much documentation about the labels and have no nice ideas for shorter names.

**Comment by AndreasBoehm** [comment]
And one more thing @schlimmchen , should we use `0.1` or `0.01` to set the battery start/stop voltage thresholds?

**Comment by schlimmchen** [comment]
> i did not fix all concerns that you mentioned

Well, in that case we can't merge this until you finally get around to fix the stuff I reported there. :grin: Thanks for pointing this out. #737 will just stay open. Simple :wink:

> should we use 0.1 or 0.01steps to set the battery start/stop voltage thresholds?

Since a step of 0.01 means the up and down arrows of the input field in HA would be unusable and since 10mV resolution for the thresholds is overkill (IMHO), we use 0.1 as you implemented.

**Comment by AndreasBoehm** [comment]
Its ready to merge 🚀

**Comment by AndreasBoehm** [comment]
I reverted the name changes as those where affecting more than just a label that is displayed.

**Pull request merged**